### PR TITLE
Dark mode and improved accessibility

### PR DESCRIPTION
**Pull request by dominik-zeglen**
I want to merge this change because it improves accessibility (increased font size and contrast) and brings brand new light and dark theme ☀️🌙

### Screenshots
Screenshot 2019-03-20 at 12 30 20
Screenshot 2019-03-20 at 12 30 29
Screenshot 2019-03-20 at 12 31 36
Screenshot 2019-03-20 at 12 31 05

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Database queries are optimized and the number of queries is constant.
1. [ ] Database migration files are up to date.
1. [ ] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [x] GraphQL schema and type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### CODE REVIEW

File: saleor/static/dashboard-next/storybook/Decorator.tsx [hunk 1]
```diff
@@ -1,19 +1,16 @@
...
 import { TimezoneProvider } from "../components/Timezone";
-import theme from "../theme";
 
 export const Decorator = storyFn => (
   <FormProvider>
     <DateProvider value={+new Date("2018-08-07T14:30:44+00:00")}>
       <TimezoneProvider value="America/New_York">
-        <MuiThemeProvider theme={theme}>
-          <CssBaseline />
+        <ThemeProvider isDefaultDark={false}>
```
Comment (maarcingebala) [inline saleor/static/dashboard-next/storybook/Decorator.tsx]: The light theme could be used as the default one, but we could also store the information about which theme is enabled in local storage and use that as default then.

File: saleor/static/dashboard-next/components/AppLayout/AppLayout.tsx [hunk 2]
```diff
@@ -12,12 +12,15 @@ import {
...
 } from "@material-ui/core/styles";
+import { CSSProperties } from "@material-ui/core/styles/withStyles";
+import Switch, { SwitchClassKey, SwitchProps } from "@material-ui/core/Switch";
 import * as classNames from "classnames";
 import * as React from "react";
 import SVG from "react-inlinesvg";
 import { RouteComponentProps, withRouter } from "react-router";
 
-import * as saleorLogo from "../../../images/logo-document.svg";
+import * as saleorLightLogo from "../../../images/logo-document.svg";
```
Comment (maarcingebala) [inline saleor/static/dashboard-next/components/AppLayout/AppLayout.tsx]: Nitpick: could we use more meaningful names for logo files e.g. something like "logo-light-theme" / "logo-dark-theme"? About the logo in the dark mode - I've noticed that it's all white while on mockups the bar over "S" should be blue.

Comment (dominik-zeglen) [reply saleor/static/dashboard-next/components/AppLayout/AppLayout.tsx]: Oopsie!
Of course we can.

File: saleor/static/dashboard-next/storybook/Decorator.tsx (hunk 1 above)
Comment (dominik-zeglen) [reply saleor/static/dashboard-next/storybook/Decorator.tsx]: This is storybook file. Theme is saved in localStorage by default.

Comment (maarcingebala) [reply saleor/static/dashboard-next/storybook/Decorator.tsx]: Right, anyway are you sure it works with localStorage? I've tested that and after closing dashboard, the theme was reset to the default light one.

File: saleor/static/dashboard-next/components/Theme/ThemeProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
+import MuiThemeProvider from "@material-ui/core/styles/MuiThemeProvider";
+import * as React from "react";
+
+import Baseline from "../../Baseline";
+import createTheme, { IThemeColors } from "../../theme";
+
+// TODO: fix secondary buttons lol
```
[the remark this answers is not in the document]
Comment (dominik-zeglen) [reply saleor/static/dashboard-next/components/Theme/ThemeProvider.tsx]: No, it now works as it is. I'll do it in the next PR.

**Comment by maarcingebala** [comment]
I reviewed the new UI with @pwgryglak today and here are a few issues we found that we could improve in this PR.

- [ ] Missing left/right margins around "Something went wrong" box:

image

- [ ] Order details page - incorrect alignment:

image

- [ ] the "more options" icon should have circular background (see the mockups on Figma)

image

- [ ] timeline section:
  - "leave comment" input looks differently on the design, it should have different background than the entire page and a border. Submit button should be placed within the input box. Also I noticed that the input isn't cleared after adding a comment.
  - increase margin above the "Order timeline" title and add a horizontal rule below it
  - timeline line and dots should be lighter - see the design

image

- [ ] Order details page - "Shipping carrier" management should be rendered in a box below addresses (I'm not sure if that's in the scope of this PR though, is not we can skip it)


image

**Comment by maarcingebala** [comment]
We could also fix this small thing:

- [ ] In the greeting on the homepage we always use user email. We should render user's first name and last name if both are available, otherwise use email.

image

**Comment by Pacu2** [comment]
Do we want to automatically detect, if a user on OS X uses dark mode?
https://drafts.csswg.org/mediaqueries-5/#prefers-color-scheme

**Pull request merged**